### PR TITLE
Adds Native Token Precompile Page

### PR DESCRIPTION
**Pull request by themacexpert**
### Description

Adds Native Token ERC-20 precompile page. 

### Checklist

- [x] I have added a label to this PR 🏷️
- [x] I have run my changes through Grammarly
- [x] If this page requires a disclaimer, I have added one
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/interact/ethereum-api/precompiles/erc20.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+Your balance will be displayed under the `balanceOf` function.
+
+### Approve a Spend {: #approve-a-spend }
+
+To approve a token spend allowance, you'll need to provide an address for the spender and the number of tokens the spender is allowed to spend. The spender can be an externally owned account (EOA) or a smart contract. For this example, you can approve the spender with an allowance of 1 UNIT token. To get started, please follow these steps:
+
+1. Expand the **approve** function
+2. Enter the address of the spender. You should have created two accounts before starting, so you can use the second account as the spender
+3. Enter the amount of tokens the spender can spend for the **value**. For this example, you can allow the spender to spend 1 UNIT token in WEI units (`1000000000000000000`)
```
Comment (eshaben) [inline builders/interact/ethereum-api/precompiles/erc20.md:137]:
```suggestion
3. Enter the amount of tokens the spender can spend for the **value**. For this example, you can allow the spender to spend 1 UNIT token in Wei units (`1000000000000000000`)
```

**Review by eshaben — COMMENTED** [review]
This stuff can be moved now! Don't forget about updating containerchain to appchain and DApp to dApp!

File: builders/interact/ethereum-api/precompiles/.pages [hunk 2]
```diff
@@ -2,4 +2,5 @@ title: Precompiles
 hide: false
 nav:
   - index.md
-  - batch.md
\ No newline at end of file
+  - batch.md
+  - erc20.md
```
Comment (eshaben) [inline builders/interact/ethereum-api/precompiles/.pages:6]: can you please use the standard 'File Title': 'file-name.md' format?

File: builders/interact/ethereum-api/precompiles/erc20.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+## The ERC-20 Solidity Interface {: #the-erc20-interface }
+
+The [`ERC20.sol`](https://github.com/moondance-labs/tanssi/blob/master/test/contracts/solidity/ERC20.sol){target=\_blank} interface on Tanssi EVM ContainerChains follows the [EIP-20 Token Standard](https://eips.ethereum.org/EIPS/eip-20){target=\_blank}, which is the standard API interface for tokens within smart contracts. The standard defines the required functions and events a token contract must implement to be interoperable with different applications.
+
+??? code "ERC20.sol"
+
+    ```solidity
+    --8<-- 'code/builders/interact/ethereum-api/precompiles/erc20/erc20.sol'
+    ```
+
```
Comment (eshaben) [inline builders/interact/ethereum-api/precompiles/erc20.md:38]:
```suggestion
```

File: builders/interact/ethereum-api/precompiles/erc20.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+The precompile is located at the following address:
+
+```text
+{{networks.dancebox.precompiles.erc20 }}
+```
+
+!!! note
+    There can be some unintended consequences when using precompiles. Tanssi's Native Token ERC-20 Precompile is derived from Moonbeam's, and as such, please familiarize yourself with [Moonbeam's Precompile Security Considerations](https://docs.moonbeam.network/builders/get-started/eth-compare/security/){target=\_blank}.
+
```
Comment (eshaben) [inline builders/interact/ethereum-api/precompiles/erc20.md:27]:
```suggestion
```

**Comment by themacexpert** [comment]
All comments addressed and base branch changed to Main

**Comment by themacexpert** [comment]
@eshaben Ready!

File: .snippets/code/builders/interact/ethereum-api/precompiles/erc20/erc20.sol [hunk 5]
```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (eshaben) [inline .snippets/code/builders/interact/ethereum-api/precompiles/erc20/erc20.sol:1]: this file can be deleted

**Review by eshaben — COMMENTED** [review]
one last thing here

File: builders/toolkit/ethereum-api/precompiles/erc20.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+To get started, open up MetaMask and make sure you are [connected to your appchain](/dapp-developers/developer-toolkit/ethereum-api/wallets/metamask/) and:
+
+1. Switch to the **Assets** tab
+2. Click on **Import tokens**
+
+![Import Tokens from Assets Tab in MetaMask](/images/builders/toolkit/ethereum-api/precompiles/erc20/erc-1.webp)
+
+Now, you can create a custom token:
+
+1. Enter the precompile address for the token contract address - `{{networks.dancebox.precompiles.erc20 }}`. When you enter the address, the **Token Symbol** and **Token Decimal** fields should automatically populate. If they do not, you can enter `UNIT` for the symbol and `18` for the decimal places. Recall that the default number of decimals for Tanssi EVM appchains is `18`, the same as Ethereum's token decimals.
```
Comment (eshaben) [inline builders/toolkit/ethereum-api/precompiles/erc20.md:60]:
```suggestion
1. Enter the precompile address for the token contract address - `{{networks.dancebox.precompiles.erc20 }}`. When you enter the address, the **Token Symbol** and **Token Decimal** fields should automatically populate. If they do not, you can enter `UNIT` for the symbol and `18` for the decimal places. Recall that the default number of decimals for Tanssi EVM appchains is `18`, the same as Ethereum's token decimals
```

File: builders/toolkit/ethereum-api/precompiles/erc20.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+One of the main benefits of this precompile is that it removes the necessity of having a wrapped representation of the protocol token as an ERC-20 smart contract, such as WETH on Ethereum. Furthermore, it minimizes the need for multiple wrapped representations of the same protocol token. Consequently, dApps that need to interact with the protocol token via an ERC-20 interface can do so without needing a separate smart contract.
+
+Under the hood, the [ERC-20 precompile](https://github.com/moondance-labs/tanssi/blob/master/test/contracts/solidity/ERC20.sol){target=\_blank} executes specific Substrate actions related to the Substrate balances module, which is coded in Rust. The balances module provides functionality for handling the various types of balances.
+
+This guide will show you how to interact with UNIT tokens, the native protocol tokens for Snap appchains on the Tanssi Dancebox Testnet, via the ERC-20 precompile. You can follow along and adapt this guide to interacting with your own appchain.
```
Comment (eshaben) [inline builders/toolkit/ethereum-api/precompiles/erc20.md:17]:
```suggestion
This guide will show you how to interact with UNIT tokens, the native protocol tokens for Snap appchains on the Tanssi Dancebox TestNet, via the ERC-20 precompile. You can follow along and adapt this guide to interacting with your own appchain.
```

**Review by eshaben — APPROVED** [review]
LGTM! Sorry for all the moves these PRs had to endure 😅

**Pull request merged**